plugins/inputs/ravendb/ravendb.go

```diff
@@ -0,0 +1,680 @@
+		}
+
+		fields := map[string]interface{}{
+			"priority":        indexResponse.Priority,
+			"state":           indexResponse.State,
+			"errors":          indexResponse.Errors,
+			"lock_mode":       indexResponse.LockMode,
+			"is_invalid":      boolToInt(indexResponse.IsInvalid),
+			"status":          indexResponse.Status,
+			"mapped_per_sec":  indexResponse.MappedPerSec,
+			"reduced_per_sec": indexResponse.ReducedPerSec,
+			"type":            indexResponse.Type,
+		}
+
+		if indexResponse.TimeSinceLastQueryInSec != nil {
+			fields["time_since_last_query_in_sec"] = indexResponse.TimeSinceLastQueryInSec
+		}
+
+		if indexResponse.TimeSinceLastIndexingInSec != nil {
+			fields["time_since_last_indexing_in_sec"] = indexResponse.TimeSinceLastIndexingInSec
+		}
+
+		acc.AddFields("ravendb_indexes", fields, tags)
+	}
+}
+
+func gatherCollections(r *RavenDB, acc telegraf.Accumulator) {
+	if !r.GatherCollectionStats {
+		return
+	}
+
+	collectionsResponse := &collectionsMetricResponse{}
+
+	err := r.requestJSON("/admin/monitoring/v1/collections"+prepareDbNamesUrlPart(r.IndexStatsDbs), &collectionsResponse)
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
+
+	for _, collectionMetrics := range collectionsResponse.Results {
+		tags := map[string]string{"url": r.URL}
+		tags["node_tag"] = collectionsResponse.NodeTag
+		tags["database_name"] = collectionMetrics.DatabaseName
+		tags["collection_name"] = collectionMetrics.CollectionName
+
+		if collectionsResponse.PublicServerUrl != "" {
+			tags["public_server_url"] = collectionsResponse.PublicServerUrl
+		}
+
+		fields := map[string]interface{}{
+			"documents_count":          collectionMetrics.DocumentsCount,
+			"total_size_in_bytes":      collectionMetrics.TotalSizeInBytes,
+			"documents_size_in_bytes":  collectionMetrics.DocumentsSizeInBytes,
+			"tombstones_size_in_bytes": collectionMetrics.TombstonesSizeInBytes,
+			"revisions_size_in_bytes":  collectionMetrics.RevisionsSizeInBytes,
+		}
+
+		acc.AddFields("ravendb_collections", fields, tags)
+	}
+}
+
+func prepareDbNamesUrlPart(dbNames []string) string {
+	if len(dbNames) == 0 {
+		return ""
+	}
+
+	result := ""
+	first := true
+
+	for _, db := range dbNames {
+		if first {
+			result += "?"
+			first = false
+		} else {
+			result += "&"
+		}
+		result += "name=" + url.QueryEscape(db)
+	}
+	return result
+}
+
+func boolToInt(b bool) int64 {
+	if b {
+		return 1
+	}
+	return 0
+}
+
+func init() {
+	inputs.Add("ravendb", func() telegraf.Input {
+		return &RavenDB{
+			ResponseHeaderTimeout: internal.Duration{Duration: DefaultResponseHeaderTimeout * time.Second},
+			ClientTimeout:         internal.Duration{Duration: DefaultClientTimeout * time.Second},
+			GatherServerStats:     true,
+			GatherDbStats:         true,
+			GatherIndexStats:      true,
+			GatherCollectionStats: true,
+			DbStatsDbs:            []string{},
+			IndexStatsDbs:         []string{},
+			CollectionStatsDbs:    []string{},
```

Comment:
This is the default anyway. Please remove.

Reply:
ok

Reply:
Thx, please mark as resolved.